release_creation/github_client/releases.py

```diff
@@ -0,0 +1,142 @@
+        if (
+            release_version.major == latest.major
+            and release_version.minor == latest.minor
+            and release_version.prerelease == latest.prerelease
+            and release_version.build == latest.build
+            and release_version.patch >= latest.patch  # type: ignore
+        ):
+            latest = release_version
+            latest_release = r
+    return latest, latest_release    
+
+
+def _get_local_snapshot_reqs(snapshot_req_path: str) -> List[str]:
+    with open(snapshot_req_path) as f:
+        reqs = f.read()
+    return reqs.split()
+
+
+def _get_gh_release_asset(release_asset: GitReleaseAsset) -> List[str]:
+    resp = requests.get(release_asset.browser_download_url)
+    resp.raise_for_status()
+    return resp.content.decode("utf-8").split()
+
+
+def _compare_reqs(snapshot_req: List[str], release_req: List[str]) -> Tuple[Set[str], Set[str]]:
+    snapshot_req_set = set(snapshot_req)
+    release_req_set = set(release_req)
+    added_req = snapshot_req_set - release_req_set
+    removed_req = release_req_set - snapshot_req_set
+    return added_req, removed_req
+
+
+def _diff_snapshot_requirements(
+    snapshot_req_path: str, latest_release: Optional[GitRelease]
+) -> str:
+    # Scenarios being handled:
+    # 1. No change - raise exception
+    # 2. No prior patch version - Creat major.minor.0 snapshot
+    # 3. New changes - generate diff
+    if latest_release:
+        diff_result = ""
+        release_reqs = [
+            _asset for _asset in latest_release.get_assets() if _SNAP_REQ_NAME in _asset.name
+        ]
+        snapshot_req = _get_local_snapshot_reqs(snapshot_req_path=snapshot_req_path)
+        release_req = _get_gh_release_asset(release_reqs[0])
+        added, removed = _compare_reqs(snapshot_req=snapshot_req, release_req=release_req)
+        if added:
+            diff_result += "Added:\n* " + "\n* ".join(added) + "\n___\n"
+        if removed:
+            diff_result += "\nRemoved:\n* " + "\n* ".join(removed) + "\n"
+        return diff_result
+    else:
+        return "No prior snapshot"
+
+
+def create_new_release_for_version(
+    release_version: Version, assets: Dict, latest_release: Optional[GitRelease]
+) -> None:
+    """Given an input version it creates a matching Github Release and attaches the assets
+       as a ReleaseAsset
+
+    Args:
+        release_version (Version): semantic version to be used when creating the release
+        assets (Dict): assets to be added to the created release where a key is the asset name
+        latest_release (Optional[GitRelease]): supply if there is a prior release to be diffed against
+
+    Raises:
+        RuntimeError: _description_
+        e: _description_
+    """
+    gh = get_github_client()
+    release_tag = str(release_version)
+    repo = gh.get_repo(_GH_SNAPSHOT_REPO)
+    reqs_files = [x for x in assets if _SNAP_REQ_NAME in x]
+    print(assets)
+    print(reqs_files)
```

Comment:
Are these left in intentionally? If so, could we add some formatting so they're more informational as logs?

Reply:
removed!